### PR TITLE
Update to build scan plugin 1.12

### PR DESCRIPTION
**Pull request by felipefzdz**

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/plugin/management/internal/autoapply/AutoAppliedBuildScanPlugin.java [hunk 1]
```diff
@@ -28,5 +28,5 @@
     PluginId ID = new DefaultPluginId("com.gradle.build-scan");
     String GROUP = "com.gradle";
     String NAME = "build-scan-plugin";
-    String VERSION = "1.11";
+    String VERSION = "1.12";
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/plugin/management/internal/autoapply/AutoAppliedBuildScanPlugin.java]: I guess we should fix the test: https://builds.gradle.org/viewQueued.html?itemId=10934610&tab=queuedBuildOverviewTab

Seem like the URL for the license changed from `ttps://scans.gradle.com/terms-of-service` to `https://gradle.com/terms-of-service`.

Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/plugin/management/internal/autoapply/AutoAppliedBuildScanPlugin.java]: This test failure seems to point to a problem with the plugin: https://builds.gradle.org/viewLog.html?buildId=10941268&tab=buildResultsDiv&buildTypeId=Gradle_Check_Stage_QuickFeedbackLinuxOnly_Trigger

WDYT?

Comment (felipefzdz) [reply subprojects/core/src/main/java/org/gradle/plugin/management/internal/autoapply/AutoAppliedBuildScanPlugin.java]: Yes, that's a legit failure. We're reusing validation logic from the non-interactive path. We'll fix it and let you know when it's ready. Thanks for having a look!

**Comment by timyates** [comment]
We released `1.12.1` of the scans-plugin which fixes the issue that was causing the test to fail

Updated the tests, and `AutoAppliedBuildScanPlugin` to use the new version

**Comment by wolfs** [comment]
I kicked off a Smoke test to see if they pass: https://builds.gradle.org/viewQueued.html?itemId=10963356&tab=queuedBuildOverviewTab

**Pull request merged**